### PR TITLE
Remove nullable disable from Processing.Processors

### PR DESCRIPTION
**Pull request by stefannikolei**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
#2231

Here I went the easy way and introduced some bangs!.

It is not easy to not use it, for example using the visitor pattern, I can not tell the compiler that the value is not null so I had to use the !

### CODE REVIEW

File: src/ImageSharp/Processing/Processors/Quantization/OctreeQuantizer{TPixel}.cs [hunk 1]
```diff
@@ -27,9 +26,9 @@ public struct OctreeQuantizer<TPixel> : IQuantizer<TPixel>
     private readonly int maxColors;
     private readonly int bitDepth;
     private readonly Octree octree;
-    private IMemoryOwner<TPixel> paletteOwner;
+    private IMemoryOwner<TPixel>? paletteOwner;
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Quantization/OctreeQuantizer{TPixel}.cs:29]: Uninitialized instances of this struct are invalid.

```suggestion
    private IMemoryOwner<TPixel> paletteOwner;
```

Rendering `!` unnecessary in all `this.paletteOwner!` expressions below.

File: src/ImageSharp/Processing/Processors/Quantization/PaletteQuantizer{TPixel}.cs [hunk 2]
```diff
@@ -21,7 +20,7 @@ namespace SixLabors.ImageSharp.Processing.Processors.Quantization;
 internal struct PaletteQuantizer<TPixel> : IQuantizer<TPixel>
     where TPixel : unmanaged, IPixel<TPixel>
 {
-    private EuclideanPixelMap<TPixel> pixelMap;
+    private EuclideanPixelMap<TPixel>? pixelMap;
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Quantization/PaletteQuantizer{TPixel}.cs:23]:
```suggestion
    private EuclideanPixelMap<TPixel> pixelMap;
```

**Review by antonfirsov — COMMENTED** [review]
Mostly good, with a few suggestions.

File: src/ImageSharp/Processing/Processors/Quantization/OctreeQuantizer{TPixel}.cs (hunk 1 above)
Comment (stefannikolei) [reply src/ImageSharp/Processing/Processors/Quantization/OctreeQuantizer{TPixel}.cs:29]: Ok then I have to remove the setting of null in dispose

**Review by JimBobSquarePants — APPROVED** [review]
Good job 👍

Some horrible internal code here that will need revisiting in the future.

**Pull request merged**